core/src/main/java/org/apache/carbondata/core/datamap/DataMapStoreManager.java

```diff
@@ -96,13 +96,19 @@ private DataMapStoreManager() {
       String dbName = carbonTable.getDatabaseName();
       String tableName = carbonTable.getTableName();
       String dmName = dataMap.getDataMapSchema().getDataMapName();
-      boolean isDmVisible = sessionInfo.getSessionParams().getProperty(
-          String.format("%s%s.%s.%s", CarbonCommonConstants.CARBON_DATAMAP_VISIBLE,
-              dbName, tableName, dmName), "true").trim().equalsIgnoreCase("true");
-      if (!isDmVisible) {
-        LOGGER.warn(String.format("Ignore invisible datamap %s on table %s.%s",
-            dmName, dbName, tableName));
-        dataMapIterator.remove();
+      if (sessionInfo != null) {
```

Comment:
please add comment to describe why this check is needed

Reply:
ok,done